src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -83,7 +83,15 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
               // between the time we retrieved them and here, where we update them.
               val allOperations = generateAttributeOperations(workspaceResponse.workspace.attributes, userAttrs,
                 k => k.namespace == AttributeName.libraryNamespace && k.name != LibraryService.publishedFlag.name)
-              rawlsDAO.patchWorkspaceAttributes(ns, name, allOperations) map (RequestComplete(_))
+              val republish = workspaceResponse.workspace.attributes.contains(publishedFlag) &&
+                workspaceResponse.workspace.attributes.get(publishedFlag).fold(false)(_.asInstanceOf[AttributeBoolean].value)
```

Comment:
my problem with get or else here, was that i couldn't just use a false as the or else because the get would return an AttributeBoolean. so my or else would have had to construct a false AttributeBoolean which seemed clunky. that's why i chose the fold, because i could execute the function on the AttributeBoolean directly if it was there and skip it if not.